reana/reana_dev/release.py

```diff
@@ -12,25 +12,127 @@
 from time import sleep
 
 import click
+import semver
 
+from reana.reana_dev.docker import docker_push
 from reana.reana_dev.git import (
     git_clean,
     git_is_current_version_tagged,
     is_last_commit_release_commit,
 )
 from reana.reana_dev.utils import (
+    display_message,
     fetch_latest_pypi_version,
     get_current_component_version_from_source_files,
+    get_current_tag,
+    is_component_dockerised,
+    parse_pep440_version,
     run_command,
     select_components,
+    translate_pep440_to_semver2,
 )
 
 
+def is_component_releasable(component, exit_code=False, display=False):
+    """Determine whether a component is releasable.
+
+    Last commit should be a release commit and the new version should be git tagged.
+
+    :param component: Component to determine whether if it is releasable or not.
+    :param exit_code: Whether the program should exit with error exit code if
+        the condition is not met.
+    :param display: Whether error messages providing instructions on how to fix
+        the problem should be displayed to stdout.
+
+    :type component: str
+    :type exit_code: bool
+    :type display: bool
+    :rtype: bool
+    """
+    is_releasable = True
+    error_message = ""
+    if not is_last_commit_release_commit(component):
+        error_message = "The last commit is not a release commit. Please use `reana-dev git-create-release-commit`."
+        is_releasable = False
+    if not git_is_current_version_tagged(component):
+        error_message = (
+            "The current version is not tagged. Please use `reana-dev git-tag`."
+        )
+        is_releasable = False
+
+    if error_message and display:
+        display_message(error_message, component)
+    if not is_releasable and exit_code:
+        sys.exit(1)
+
+    return is_releasable
+
+
 @click.group()
 def release_commands():
     """Release commands group."""
 
 
+@click.option(
+    "--component",
+    "-c",
+    required=True,
+    multiple=True,
+    help="Which components? [name|CLUSTER]",
+)
+@click.option("--user", "-u", default="reanahub", help="DockerHub user name [reanahub]")
+@click.option("--image-name", help="Should the component have a custom image name?")
+@release_commands.command(name="release-docker")
+@click.pass_context
+def release_docker(ctx, component, user, image_name):  # noqa: D301
+    """Release a component on dockerhub.com.
```

Comment:
Cosmetics: "on Docker Hub", or "on hub.docker.com" (which is the canonically visible URL).